tests/Functional/PreventOverlappingTest.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace Crunz\Tests\Functional;
```

Comment:
This test seems to be `unit`, but it is in `Functional` namespace, it is intended?

Reply:
You are right. At first I thought it's gonna be more of a functional test, I'll move it.